docs/subnets/create-a-vm-blobvm.md

```diff
@@ -0,0 +1,454 @@
+# Create a Blob Virtual Machine (VM)
+
+[toc]
+
+## Introduction
+
+Virtual Machine is like a blueprint for a blockchain and blockchain instantiates from VM, very much similar to a class-object relationship. Issuance of transactions, transaction types, block structure, block building algorithm, keeping pending transactions in the mempool, gossiping transactions to the connected nodes, etc. are defined in a virtual machine.
+
+Virtual Machines can be broken down into components like **Transaction**, **Mempool**, **Network**, **Block**, etc. Each component works together to -
+* Define the representation of the blockchain's state
+* Represent the operations in that state
+* Apply the operations in that state
+
+## BlobVM
+
+In this tutorial, we will learn all about virtual machines and how we can build them by taking references from **[BlobVM](https://github.com/ava-labs/blobvm)**. It's a virtual machine that can be used to instantiate key-value blockchains for storing files like images, videos, etc. in an efficient way.
+
+Blobs are small chunks of data. In BlobVM, we divide a file into small blobs and store them as a key-value pair. The key of these small chunks of a file is linked together as the children of the root. This tree is also stored as JSON data against the root's key.
+
+## How to use a VM
+
+VM modules are served as a binary plugin to the `avalanchego` software. The nodes that want to power their chains using a particular VM, must have the built binary in their avalanchego's `build/plugins` directory. There could be multiple VM plugins in this directory.
+
+## High-Level Overview
+
+Virtual Machines are created as a module, whose binary is registered (loaded) by a node running `avalanchego`, against a user-defined **vmID** (binary file name must be vmID). We can build multiple chains with the registered VM. A VM is registered, only when its built binary is present in the `avalanchego/build/plugins/` directory.
+
+The VM exposes 3 things for interaction -
+* **Static and Non-Static Handlers (API)** for making requests to VM methods
+* **Service** for implementing handlers or APIs
+* **Client** for making calls to the service's APIs
+
+### What are Handlers
+
+Handlers serve the response for the incoming HTTP requests. Handlers can also be wrapped with **gRPC** for efficiently making calls from other services such as avalanchego. Handler helps in creating APIs. VM implements 2 kinds of handlers -
+
+* **Non-Static Handlers** - They help in interacting with blockchains instantiated by the VM. The API's endpoint will be different for different chains. `/ext/bc/[chainID]`
+* **Static Handlers** - They help in directly accessing VM. These are optional for reasons such as parsing genesis bytes required to instantiate new blockchains. `/ext/vm/[vmID]`
+
+### Registering a VM
```

Comment:
I thought VM registry was something the user didn't care about (handled by avalanchego under-the-hood). Should this be in this tutorial? I don't see this in the `blobvm` code.

Reply:
One suggestion: put what users must implement on the top, and use appendix for detailed explanation at the end of the doc.

Reply:
I think the chain manager itself may be better suited to a separate document or at the very least to an appendix/bottom of the doc as @yulin-dong  suggested.

Reply:
This feels like an unnecessary part of the doc imo, I would be fine removing it. We can just tell users that they need to implement the factory interface, but that's already called out in https://docs.avax.network/subnets/create-a-virtual-machine-vm#factory. I feel like letting users know what's going on under-the-hood is unnecessary for people trying to build VMs and will just be more stuff we need to update as the codebase evolves.